nav2_util/include/nav2_util/lifecycle_utils.hpp

```diff
@@ -50,6 +50,33 @@ void bringup_lifecycle_nodes(
   bringup_lifecycle_nodes(split(nodes, ':'), service_call_timeout, retries);
 }
 
+/// Transition the given lifecycle nodes to the UNCONFIGURED state in order
+/** At this time, service calls frequently hang for unknown reasons. The only
+ *  way to combat that is to timeout the service call and retry it. To use this
+ *  function, estimate how long your nodes should take to at each transition and
+ *  set your timeout accordingly.
+ * \param[in] node_names A vector of the fully qualified node names to bringup.
+ * \param[in] service_call_timeout The maximum amount of time to wait for a
+ *            service call.
+ * \param[in] retries The number of times to try a state transition service call
+ */
+void bringdown_lifecycle_nodes(
```

Comment:
Should 'bringdown' be 'shutdown'? We use shutdown in the service call to lifecycle_manager. I think it makes sense to be consistent.

Reply:
Plus shutdown matches the lifecycle state machine terminology

Reply:
Maybe 'bringup' should also be 'startup'?

Reply:
> Should 'bringdown' be 'shutdown'? We use shutdown in the service call to lifecycle_manager. I think it makes sense to be consistent.

> Plus shutdown matches the lifecycle state machine terminology

@mkhansen-intel I'm not sure. Looking at the [diagram](https://design.ros2.org/img/node_lifecycle/life_cycle_sm.png), `shutdown` takes the node to `finalize` state, while this method takes the node back to `unconfigured` state -- assuming is on `active`. Is there another term we can use?

Reply:
> Maybe 'bringup' should also be 'startup'?

@mkhansen-intel That's possible since the diagram doesn't show a term for the process of taking a node from `active` to `unconfigured`.

Reply:
OK, so I agree that "shutdown" is different. How about more descriptive names:
bringup -> configure_activate
bringdown -> deactivate_cleanup

I know it's a little more wordy but it's clear from the name what it means and avoids confusion with 'shutdown'.

I'm also OK with changing these to 'pause' and 'resume' as I believe we're using those terms to mean these same actions in other header files. At least we'd be self-consistent then.

Reply:
If those are the options, I'd actually prefer to go with the long descriptive names `configure_activate` and `deactivate_cleanup`. I'm not sure we have a good definition yet for `pause` or `resume`.

Reply:
I think that would make sense if the actions occuring fit into those scopes. I might add an `_and_` between them to be really clear but up to you